lib/cache/ResolverCachePlugin.js

```diff
@@ -16,29 +16,52 @@ const makeSerializable = require("../util/makeSerializable");
 
 class CacheEntry {
 	constructor(
-		result,
+		resolveResult,
 		fileDependencies,
 		contextDependencies,
 		missingDependencies,
 		snapshot
 	) {
-		this.result = result;
+		this.result = resolveResult;
 		this.fileDependencies = fileDependencies;
 		this.contextDependencies = contextDependencies;
 		this.missingDependencies = missingDependencies;
 		this.snapshot = snapshot;
 	}
 
-	serialize({ write }) {
-		write(this.result);
+	serialize({ write, writePath }) {
+		if (!this.result) {
+			write(this.result);
+		} else {
+			const {
+				descriptionFilePath,
+				descriptionFileRoot,
+				path,
+				...rest
+			} = this.result;
+
+			write(rest);
+			writePath(descriptionFilePath);
+			writePath(descriptionFileRoot);
+			writePath(path);
+		}
+
 		write(this.fileDependencies);
 		write(this.contextDependencies);
 		write(this.missingDependencies);
 		write(this.snapshot);
 	}
 
-	deserialize({ read }) {
-		this.result = read();
+	deserialize({ read, readPath }) {
+		let result = read();
+
+		if (result) {
+			result.descriptionFilePath = readPath();
+			result.descriptionFileRoot = readPath();
+			result.path = readPath();
+		}
```

Comment:
`resolveResult` provided by `enhanced-resolve`, did not find better way to (de)serialize it.. Probably could lead to deop since `resolveResult` from cache could have another hidden class..